packages/suite/src/components/suite/layouts/SuiteLayout/AppShortcuts.tsx

```diff
@@ -1,15 +1,19 @@
 import { useEvent } from 'react-use';
 
 import { selectSelectedDevice } from '@suite-common/wallet-core';
+import { WalletType } from '@suite-common/wallet-types';
 import { KEYBOARD_CODE } from '@trezor/components';
 
 import { closeModalApp, goto } from 'src/actions/suite/routerActions';
 import { addWalletThunk } from 'src/actions/wallet/addWalletThunk';
 import { useDiscovery, useDispatch, useSelector } from 'src/hooks/suite';
 
-export const useAppShortcuts = () => {
+import { usePassphraseModalContext } from '../../modals/ReduxModal/DeviceContextModal/PassphraseModalContext';
+
+export const AppShortcuts = () => {
     const selectedDevice = useSelector(selectSelectedDevice);
     const dispatch = useDispatch();
+    const { setPassphraseState, setIsExisting } = usePassphraseModalContext();
```

Comment:
This context keeps the passphrase state and identification if we went through isExisting branch in the state machine. Theoretically this info can be extracted also from passphraseState.

E.g.: `not-exist-enter-passphrase` state should have `isExisting = false`